azure-nextgen-static-website/index.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021, Pulumi Corporation.  All rights reserved.
+
+import * as cdn from "@pulumi/azure-nextgen/cdn/latest";
+import * as resources from "@pulumi/azure-nextgen/resources/latest";
+import * as storage from "@pulumi/azure-nextgen/storage/latest";
+import * as pulumi from "@pulumi/pulumi";
+import * as random from "@pulumi/random";
+
+// TODO: Remove after autonaming support is added.
+const randomSuffix = new random.RandomString("random", {
+    length: 10,
+    special: false,
+    upper: false,
+});
+
+const resourceGroup = new resources.ResourceGroup("resourceGroup", {
+    resourceGroupName: pulumi.interpolate `rg${randomSuffix.result}`,
+    location: "westus",
+});
+
+const config = new pulumi.Config();
+const storageAccountName = config.get("storageAccountName") || pulumi.interpolate `site${randomSuffix.result}`;
+const cdnEndpointName = config.get("cdnEndpointName") || pulumi.interpolate `cdn-endpnt-${storageAccountName}`;
+const cdnProfileName = config.get("cdnProfileName") || pulumi.interpolate `cdn-profile-${storageAccountName}`;
+
+const profile = new cdn.Profile("profileResource", {
+    profileName: cdnProfileName,
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        name: "Premium_Verizon",
+    },
+    tags: {},
+});
+
+const storageAccount = new storage.StorageAccount("storageAccount", {
+    accessTier: "Hot",
+    accountName: storageAccountName,
+    enableHttpsTrafficOnly: true,
+    encryption: {
+        keySource: "Microsoft.Storage",
+        services: {
+            blob: {
+                enabled: true,
+            },
+            file: {
+                enabled: true,
+            },
+        },
+    },
+    kind: storage.Kind.StorageV2,
+    networkRuleSet: {
+        bypass: "AzureServices",
+        defaultAction: "Allow",
+        ipRules: [],
+        virtualNetworkRules: [],
+    },
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        name: "Standard_LRS",
+    },
+    tags: {},
+});
+
+const endpointOrigin = storageAccount.primaryEndpoints.apply(ep => ep.web.replace("https://", "").replace("/", ""));
+
+const endpoint = new cdn.Endpoint("CDNEndpoint", {
+    contentTypesToCompress: [],
+    endpointName: cdnEndpointName,
+    isCompressionEnabled: false,
+    isHttpAllowed: true,
+    isHttpsAllowed: true,
+    originHostHeader: endpointOrigin,
+    origins: [{
+        hostName: endpointOrigin,
+        httpPort: 80,
+        httpsPort: 443,
+        name: pulumi.interpolate `${cdnEndpointName}-origin-${randomSuffix.result}`,
+    }],
+    profileName: profile.name,
+    queryStringCachingBehavior: "NotSet",
+    resourceGroupName: resourceGroup.name,
+    tags: {},
```

Comment:
I think we can remove a bunch of irrelevant properties

Reply:
Might have to do with the arm template I started with. Cleaned up to a large extent now.